autorest/autorest_test.go

```diff
@@ -0,0 +1,79 @@
+package autorest
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+
+	"github.com/Azure-Samples/azure-sdk-for-go-samples/helpers"
+	"github.com/Azure-Samples/azure-sdk-for-go-samples/iam"
+	"github.com/Azure/azure-sdk-for-go/services/storage/mgmt/2017-06-01/storage"
+	"github.com/Azure/go-autorest/autorest"
+	"github.com/Azure/go-autorest/autorest/to"
+)
+
+var accountName = "testname01"
+
+func requestInspect() autorest.PrepareDecorator {
+	return func(p autorest.Preparer) autorest.Preparer {
+		return autorest.PreparerFunc(func(r *http.Request) (*http.Request, error) {
+			log.Printf("Inspecting Request: %s %s\n", r.Method, r.URL)
+			return p.Prepare(r)
+		})
+	}
+}
+
+func respondInspect() autorest.RespondDecorator {
+	return func(r autorest.Responder) autorest.Responder {
+		return autorest.ResponderFunc(func(resp *http.Response) error {
+			log.Printf("Inspecting Response: %s for %s %s\n", resp.Status, resp.Request.Method, resp.Request.URL)
+			return r.Respond(resp)
+		})
+	}
+}
```

Comment:
[debate topic]

Instead of having a getter for a static method, I'd be just as happy to declare this function directly:

``` Go
func loggingResponseInspector(r autorest.Responder) autorest.Responder {
    return autorest.ResponderFunc(func(resp.*http.Response) error {
        log.Printf("Inspecting Response: %s for %s %s\n", resp.Status, resp.Request.Method, resp.Request.URL)
        return r.Respond(resp)
    }
}
```

If we keep it in the current form, I'd suggest trying to make it demonstrative by adding in something to explicitly take advantage of the closure: (Notice the injected logger)

``` Go
func respondInspect(l *log.Logger) autorest.RespondDecorator {
	return func(r autorest.Responder) autorest.Responder {
		return autorest.ResponderFunc(func(resp *http.Response) error {
			l.Printf("Inspecting Response: %s for %s %s\n", resp.Status, resp.Request.Method, resp.Request.URL)
			return r.Respond(resp)
		})
	}
}
```

Reply:
The same comment would apply for the request side of this.